### DISCUSSION TITLE
OpenAPI v 3.0.X invalid when model can be None

### DISCUSSION THREAD

**Discussion by panfilenok-epam** [fastapi/fastapi#10953] [Questions]
### First Check

- [x] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.encoders import jsonable_encoder
from fastapi.responses import JSONResponse
from fastapi.openapi.utils import get_openapi
from pydantic import BaseModel

class Item(BaseModel):
    name: str

class Model(BaseModel):
    item: Item | None

app = FastAPI()

def custom_openapi():
    if app.openapi_schema:
        return app.openapi_schema
    openapi_schema = get_openapi(
        title="Test",
        version="1.0.0",
        openapi_version="3.0.2",
        routes=app.routes,
    )

    app.openapi_schema = openapi_schema
    return app.openapi_schema

app.openapi = custom_openapi

@app.get("/", response_model=Model)
async def get() -> JSONResponse:
    return JSONResponse(content=jsonable_encoder(Model(item=None)), status_code=200)
```

### Description

The problem is when an OpenAPI document is generated it contains the following:
```
...{"properties":{"item":{"anyOf":[{"$ref":"#/components/schemas/Item"},{"type":"null"}]}}...
```
which appears to be invalid for the document schema v3.0.x
`Structural error at components.schemas.Model.properties.item.anyOf.1.type
should be equal to one of the allowed values
allowedValues: array, boolean, integer, number, object, string`

### Operating System

Linux

### FastAPI Version

0.109.0

### Pydantic Version

2.5.2

### Python Version

3.11.7

**Comment by JavierSanchezCastro** [comment]
It seems a bit weird, but if I run the code with your ``custom_openapi`` and a test endpoint, I get an error since you omitted the ``title`` parameter in ``get_openapi``.

In another case, try to show and upload an MRE we can just run.

**Reply by panfilenok-epam** [reply]
thank you for the fast reply. I have updated the code and the error message

**Reply by panfilenok-epam** [reply]
Any update?

**Comment by YuriiMotov** [comment]
I think the `openapi_version` parameter of `get_openapi` is misleading..

Looking at source code I see that it's only used as a value of `openapi` field in output schema. It doesn't affect schema generation in any other way except just replacing this field in the resulting schema.
https://github.com/fastapi/fastapi/blob/e31f35ec65c263f9cde2c7c7fa4603577e6b8392/fastapi/openapi/utils.py#L477-L569

So, the result of `get_openapi` is always schema that follows OAS 3.1, no matter what is specified in `openapi` field.
